inference-engine/src/preprocessing/cpu_x86_avx2/ie_preprocess_gapi_kernels_avx2.cpp

```diff
@@ -147,75 +468,35 @@ void calcRowLinear_8UC_Impl(std::array<std::array<uint8_t*, 4>, chanNum> &dst,
-                                       0, 4, 8, 12, 2, 6, 10, 14,
-                                       1, 5, 9, 13, 3, 7, 11, 15);
-
-        v_uint8 shuf_mask3 = v_setr_s8(0, 1, 8,  9,  2, 3, 10, 11,
-                                       4, 5, 12, 13, 6, 7, 14, 15,
-                                       0, 1, 8,  9,  2, 3, 10, 11,
-                                       4, 5, 12, 13, 6, 7, 14, 15);
-
         // vertical pass
         v_int16 b0 = vx_setall_s16(beta[0]);
         v_int16 b1 = vx_setall_s16(beta[1]);
         v_int16 b2 = vx_setall_s16(beta[2]);
         v_int16 b3 = vx_setall_s16(beta[3]);
 
-        for (int w = 0; w < inSz.width*chanNum; ) {
-            for (; w <= inSz.width*chanNum - half_nlanes && w >= 0; w += half_nlanes) {
-                v_int16 val0_0 = v_load_ccache_expand(&src0[0][w]);
-                v_int16 val0_1 = v_load_ccache_expand(&src0[1][w]);
-                v_int16 val0_2 = v_load_ccache_expand(&src0[2][w]);
-                v_int16 val0_3 = v_load_ccache_expand(&src0[3][w]);
-
-                v_int16 val1_0 = v_load_ccache_expand(&src1[0][w]);
-                v_int16 val1_1 = v_load_ccache_expand(&src1[1][w]);
-                v_int16 val1_2 = v_load_ccache_expand(&src1[2][w]);
-                v_int16 val1_3 = v_load_ccache_expand(&src1[3][w]);
-
-                v_int16 t0 = v_mulhrs(v_sub_wrap(val0_0, val1_0), b0);
-                v_int16 t1 = v_mulhrs(v_sub_wrap(val0_1, val1_1), b1);
-                v_int16 t2 = v_mulhrs(v_sub_wrap(val0_2, val1_2), b2);
-                v_int16 t3 = v_mulhrs(v_sub_wrap(val0_3, val1_3), b3);
-
-                v_int16 r0 = v_add_wrap(val1_0, t0);
-                v_int16 r1 = v_add_wrap(val1_1, t1);
-                v_int16 r2 = v_add_wrap(val1_2, t2);
-                v_int16 r3 = v_add_wrap(val1_3, t3);
-
-                v_uint8 q0 = v_packus(r0, r1);
-                v_uint8 q1 = v_packus(r2, r3);
-
-                v_uint8 q2 = v_blend_shiftleft<0xCC /*0b11001100*/, 4>(q0, q1);
-                v_uint8 q3 = v_blend_shiftright<0xCC /*0b11001100*/, 4>(q0, q1);             
-
-                v_uint8 q4 = v_shuffle_s8(q2, shuf_mask1);
-                v_uint8 q5 = v_shuffle_s8(q3, shuf_mask1);
-
-                v_uint8 q6 = v256_permute2x128<0x20>(q4, q5);
-                v_uint8 q7 = v256_permute2x128<0x31>(q4, q5);
-
-                vx_store(&tmp[4 * w + 0],  q6);
-                vx_store(&tmp[4 * w + 2 * half_nlanes], q7);
-            }
-
-            if (w < inSz.width*chanNum) {
-                w = inSz.width*chanNum - half_nlanes;
-            }
-        }
+        verticalPass_lpi4_8U(src0, src1, tmp, b0, b1, b2, b3, shuf_mask1,
+                             inSz.width*chanNum, half_nlanes);
 
         // horizontal pass
-        v_uint8 val_0, val_1, val_2, val_3;
         GAPI_DbgAssert(outSz.width >= half_nlanes);
+        v_uint8 val_0, val_1, val_2, val_3, res1, res2;
+
+        v_uint8 shuf_mask2 = v_setr_s8(0, 4, 8, 12, 2, 6, 10, 14,
```

Comment:
Please make this v_setr_s8 a function (v_uint8 hor_shufle_mask0()) and reuse it in horizontalPass_lpi4_8UC1 when masks are moved there

Reply:
Done.